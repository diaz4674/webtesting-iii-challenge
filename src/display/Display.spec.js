import React from 'react';
import renderer from 'react-test-renderer'; // 1: install this npm module as a dev dependency

import {render, cleanup, fireEvent} from '@testing-library/react'
import '@testing-library/react/cleanup-after-each'
import Display from './Display'
afterEach(cleanup)

describe('<Display  />', () => {
    it('matches snapshot', () => {
        const tree = renderer.create(<Display/>)
        expect(tree.toJSON()).toMatchSnapshot()
    })

    it('should display closed if closed prop is true', () =>{

        const {getByText, queryByText} = render(<Display closed = {true}/>)

        getByText(/Closed/i)
    })

    it('should display open if closed prop is false', () => {
        const {getByText, queryByText} = render(<Display/>)

        getByText(/Open/i)

    })

    it('should display Locked if locked prop is true', () => {
        const {getByText, queryByText} = render(<Display locked={true}/>)

        getByText(/Locked/i)
    })

    it('should display Unlocked if locked prop is false', () => {
        const {getByText,  queryByText} = render(<Display locked={false}/>)

        getByText(/Unlocked/i)
    })

    it("when locked or closed use the red-led class", () => {
        const display = render(<Display locked = {true} closed = {true} />)
        const lockedDiv = display.getByTestId(/locked-div/i)
        const closedDiv = display.getByTestId(/closed-div/i)

        expect(lockedDiv.classList[1]).toBe('red-led')
        expect(closedDiv.classList[1]).toBe('red-led')
    })

    it('closes panel sets classname to green-led when open or unlocked', () => {
        const display = render(<Display locked = {false} closed = {false} />)
        
        const lockedDiv = display.getByTestId(/locked-div/i)
        const closedDiv = display.getByTestId(/closed-div/i)

        expect(lockedDiv.classList[1]).toBe('green-led')
        expect(closedDiv.classList[1]).toBe('green-led')
    })
})