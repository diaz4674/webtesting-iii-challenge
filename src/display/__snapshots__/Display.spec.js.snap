// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display  /> matches snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
    data-testid="locked-div"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
    data-testid="closed-div"
  >
    Open
  </div>
</div>
`;
